src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -88,30 +95,55 @@ def _get_email_domain(email: str) -> str | None:
         return None
 
     def get(self, request: Request, organization: Organization) -> Response:
-        queryset = self._get_missing_members(organization)
+        # ensure the organization has an integration with the commit feature
+        integrations = integration_service.get_integrations(
+            organization_id=organization.id, status=ObjectStatus.ACTIVE
+        )
+
+        integrations_with_commits = [
+            i for i in integrations if i.has_feature(feature=IntegrationFeatures.COMMITS)
+        ]
+        integration_providers = {i.provider for i in integrations_with_commits}
```

Comment:
You can avoid the multiple loops here with a dictionary and reducing the initial integrations query:
```python
from functools import reduce

def provider_reducer(dict, integration):
  if not integration.has_feature(....):
    return dict
  if dict.get(integration.provider):
    dict[integration.provider].append(integration.id)
    return dict
  dict[integration.provider] = [integration.id]
    

integration_provider_to_ids = reduce(provider_reducer, integrations, {})
```
or something like that at least. It'll prevent having to iterate through `integrations_with_commits` again while looking at each provider.

Reply:
this is very fancy :O